### ISSUE TITLE
take field/method name as default id

### ISSUE THREAD

**Issue by SierraGolf** [JakeWharton/butterknife#123]
Hi,

tl;dr... What do you think about supporting field/method names as "clever" default identifiers?

I am currently experimenting how I can speed up the build process during development for a rather large android project which uses otto and butterknife. The greatest speed up  is probably not cleaning the whole working directory before re-building the project, thus using incremental compilation and dexing. This kind of works pretty well as long as nobody touches ids (R.id.xxx).

Since ids are by default constants (static final), they are inlined at compile time and break incremental compilation when changed (app does not find the right objects at runtime). One way to circumvent this is passing "--non-constant-id" argument to the aapt tool for development builds. As a result ids are still static but not final anymore, which in return makes them not usable as annotation parameters.

So it would be great if one could optionally omit the explicit reference to the id in the annotations. 

The only thing that immediately comes to my mind, which could cause some trouble when implementing this would be ambiguity with multiple listener annotations for the same id. 

What do you think?
